Sources/Auth/Core/SocialAuthenticator.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  SocialAuthenticator.swift
+//  PovioKit
+//
+//  Created by Borut Tomazin on 30/01/2023.
+//  Copyright © 2023 Povio Inc. All rights reserved.
+//
+
+import PovioKitPromise
+import Foundation
+import UIKit
+
+public class SocialAuthenticator {
+  private let authenticators: [Authenticator]
+  
+  public init(authenticators: [Authenticator]) {
+    self.authenticators = authenticators
+  }
+}
+
+extension SocialAuthenticator {
+  public func authenticator<A: Authenticator>(for type: A.Type) -> A? {
+    authenticators.first { $0 is A } as? A
+  }
+  
+  public func signOut() {
```

Comment:
This should return a promise. Another implementation would be to just call `signOut` on all the authenticators.

Reply:
Why?